### ISSUE TITLE
Problem about the convergence of gmm loss 

### ISSUE THREAD

**Issue by 31sy** [Luffy03/AGMM-SASS#9]
Hi. thanks for sharing the code. I have tried to run the AGMM on the Pascal VOC dataset. However, I find the gmm loss cannot be trained tending to convergence. The mIoU just achieves 60, which is lower than the paper. Can you help analyze the reason? Or how can I solve it? Thanks.

My environment is Ubuntu 20, two 3090 gpus, pytorch 1.8.0, cuda11, batch size is 10 per gpu.

The  loss log is:
[2024-08-14 10:46:53,957][    INFO] Iters:0, loss:1.580, seg_loss:0.209, gmm_loss:1.286, proto_loss:0.032, cls_loss:0.053
[2024-08-14 10:48:04,257][    INFO] Iters:66, loss:3.102, seg_loss:0.262, gmm_loss:2.789, proto_loss:0.030, cls_loss:0.021
[2024-08-14 10:49:14,297][    INFO] Iters:132, loss:3.086, seg_loss:0.251, gmm_loss:2.785, proto_loss:0.025, cls_loss:0.025
[2024-08-14 10:50:24,178][    INFO] Iters:198, loss:3.099, seg_loss:0.239, gmm_loss:2.810, proto_loss:0.025, cls_loss:0.025
[2024-08-14 10:51:33,648][    INFO] Iters:264, loss:3.058, seg_loss:0.235, gmm_loss:2.771, proto_loss:0.025, cls_loss:0.027
[2024-08-14 10:52:43,014][    INFO] Iters:330, loss:3.033, seg_loss:0.237, gmm_loss:2.742, proto_loss:0.025, cls_loss:0.028
[2024-08-14 10:53:53,058][    INFO] Iters:396, loss:3.059, seg_loss:0.240, gmm_loss:2.765, proto_loss:0.025, cls_loss:0.028
[2024-08-14 10:55:02,676][    INFO] Iters:462, loss:3.058, seg_loss:0.239, gmm_loss:2.765, proto_loss:0.026, cls_loss:0.028
[2024-08-14 10:56:12,544][    INFO] Iters:528, loss:3.035, seg_loss:0.234, gmm_loss:2.748, proto_loss:0.025, cls_loss:0.027

**Comment by Luffy03** [comment]
Hi, many thanks for your attention to our work. 
It is the first time I have seen such training logs. What settings do you conduct? Point- or scribble-supervised?

**Comment by 31sy** [comment]
I use the point-supervised setting. I separate the original gmm loss into three parts. I find the error loss is in the gmm part. The other two losses are correct.

**Comment by Luffy03** [comment]
Sorry for my late reply. I have not yet met the same problems. I am not sure about the reasons now, maybe it is caused by the torch or cuda version. This project is done about two years ago, maybe changing them to old versions will work. Let me check my training logs again.

**Comment by 31sy** [comment]
Thanks for your reply. I check the label file. I find the error is that I misuse the point-label. I use my voc point label(just one point for one class) for training, But the point label in the TEL has more points for each class. I have changed it, and the gmm loss is normal now. Thanks for your help again.

**Comment by Luffy03** [comment]
Wow, many thanks! Glad to hear that!
